### PR TITLE
MS Edge compatibility

### LINKED ISSUE #34: Request - Edge Browser extension
While I do use Chrome, I prefer to use Edge on Windows 10.
I'd love to see this extension also available for Edge browser.

### PR DESCRIPTION
**Pull request by mazmazz**
Closes #34. Edge loads this extension with the following changes:

* `manifest.json`: Add `author` field
* `manifest.json`: Add `background.persistent = false`
* `background.js`: Define `chrome.webRequest` by pointing to `browser.webRequest`

It sideloads and works fine in all three browsers. 

Packaging and submission info is [here](https://docs.microsoft.com/en-us/microsoft-edge/extensions/getting-started#packaging-an-extension); ~~I haven't looked into this personally.~~ (edit: see [comment](https://github.com/nextgens/anti-paywall/pull/103#issuecomment-426148279), it works)

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -145,6 +145,11 @@ function blockCookies(details) {
 	return {responseHeaders: responseHeaders};
 }
 
+// MS Edge support
+if (!chrome.webRequest) {
+  chrome.webRequest = browser.webRequest;
+}
```
Comment (mazmazz) [inline background.js:151]: I think this is the best polyfill because of this compat weirdness:

* Edge defines `window.chrome` already, but not `chrome.webRequest`.
* Edge defines `browser.webRequest` as callback-based. 
* Firefox defines `chrome.webRequest` as callback-based and `browser.webRequest` as Promise-based.
* Chrome defines `chrome.webRequest` as callback-based and does NOT define `browser.webRequest`.

So I could not directly use `browser.webRequest` because Chrome does not have it and Firefox implements it differently. I also could not do the single-line `chrome.webRequest = chrome.webRequest || browser.webRequest` because Chrome evaluates both expressions and errors out.

File: manifest.json [hunk 2]
```diff
@@ -1,10 +1,12 @@
 {
   "name": "Anti-Paywall",
+  "author": "nextgens",
   "version": "1.0",
   "description": "This extension will mangle your browser's requests to maximize the chances of bypassing paywalls.",
   "permissions": ["webRequest", "webRequestBlocking", "*://*/*"],
   "background": {
-    "scripts": ["background.js"]
+    "scripts": ["background.js"],
+    "persistent": false
```
Comment (mazmazz) [inline manifest.json:9]: `background.persistent` is required for Edge. Chrome says to [use `false`](https://developer.chrome.com/extensions/background_pages#manifest) for background scripts (and I assume this is default).

Firefox outputs this warn message upon side-loading, but otherwise the extension works fine. It comes across as an error, but is actually a warn. [See Bugzilla issue on changing this message](https://bugzilla.mozilla.org/show_bug.cgi?id=1370077). I don't think this affects the user install experience, but I'm unfamiliar with that process.

image

**Comment by mazmazz** [comment]
Edge extension packaging works. See test package: [anti-paywall-9044fce.zip](https://github.com/nextgens/anti-paywall/files/2436500/anti-paywall-9044fce.zip)

I followed the instructions here:

* [Using ManifoldJS to create extension AppX packages](https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/packaging/using-manifoldjs-to-package-extensions)
    * More info about Dev Center identity keys: [Creating and testing a Microsoft Edge extension AppX package](https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/packaging/creating-and-testing-extension-packages)
* [Create a certificate for package signing](https://docs.microsoft.com/en-us/windows/uwp/packaging/create-certificate-package-signing) *(You will not need to do this if submitting to the Windows Store)*
* [Sign an app package using SignTool](https://docs.microsoft.com/en-us/windows/uwp/packaging/sign-app-package-using-signtool) *(Ditto)*

For submission, you would need to provide the app icon at 44x44, 50x50, and 150x150. You would also need to sign up with Windows Dev Center and input the supplied Identity keys in the packaging step. See first link for more details.
